src/Components/Search/Spike/SearchBarSpike.tsx

```diff
@@ -0,0 +1,226 @@
+      leading: true,
+    })
+    this.throttledOnSuggestionHighlighted = throttle(
+      this.throttledOnSuggestionHighlighted,
+      500,
+      { leading: true }
+    )
+  }
+
+  searchTextChanged = (_e, { newValue: input }) => {
+    this.setState({ input })
+  }
+
+  onSuggestionsClearRequested = () => {
+    this.setState({ input: "", entityID: null, entityType: null })
+  }
+
+  // Navigate to selected search item.
+  onSuggestionSelected = (
+    _e,
+    {
+      suggestion: {
+        node: { href },
+      },
+    }
+  ) => {
+    window.location.href = href
+  }
+
+  renderPreview() {
+    const { entityID, entityType } = this.state
+    if (entityID && entityType) {
+      return <SearchPreview entityID={entityID} entityType={entityType} />
+    }
+  }
+
+  renderSuggestionsContainer = ({ containerProps, children, query }) => {
+    if (!query) return null
+    return (
+      <Box {...containerProps}>
+        <Flex flexDirection={["column", "row"]}>
+          <Box width={["100%", "25%"]}>
+            <Flex flexDirection="column">
+              <Box>Search "{query}"</Box>
+              {children}
+            </Flex>
+          </Box>
+          <Box width={["100%", "75%"]}>{this.renderPreview()}</Box>
+        </Flex>
+      </Box>
+    )
+  }
+
+  renderSuggestion = ({ node: { displayLabel } }, { isHighlighted }) => {
+    return (
+      <span style={{ backgroundColor: isHighlighted ? "#ddd" : "#fff" }}>
+        {displayLabel}
+      </span>
+    )
+  }
+
+  renderInputComponent = inputProps => (
+    <Box>
+      <Input style={{ width: "100%" }} {...inputProps} />
+    </Box>
+  )
+
+  render() {
+    const { viewer } = this.props
+    const { input } = this.state
+    const edges = get(viewer, v => v.search.edges, [])
+
+    const inputProps = {
+      onChange: this.searchTextChanged,
+      placeholder: "Search by artist, gallery, style...",
+      value: input,
+    }
+
+    return (
+      <Autosuggest
+        suggestions={edges}
+        onSuggestionsClearRequested={this.onSuggestionsClearRequested}
+        onSuggestionHighlighted={this.throttledOnSuggestionHighlighted}
+        onSuggestionsFetchRequested={this.throttledFetch}
+        getSuggestionValue={({ node: { displayLabel } }) => displayLabel}
+        renderSuggestion={this.renderSuggestion}
+        renderSuggestionsContainer={this.renderSuggestionsContainer}
+        inputProps={inputProps}
+        onSuggestionSelected={this.onSuggestionSelected}
+        renderInputComponent={this.renderInputComponent}
+      />
+    )
+  }
+}
+
+export const SearchBarRefetchContainer = createRefetchContainer(
+  SearchBarSpike,
+  {
+    viewer: graphql`
+      fragment SearchBarSpike_viewer on Viewer
+        @argumentDefinitions(
+          term: { type: "String!", defaultValue: "" }
+          hasTerm: { type: "Boolean!", defaultValue: false }
+        ) {
+        search(query: $term, mode: AUTOSUGGEST, first: 10)
+          @include(if: $hasTerm) {
```

Comment:
Do we get a performance advantage by including the `edges.node....` conditionally based on `hasTerm`? If not, we get zero results when passing in an empty string, and we could remove the `hasTerm` argument.

Reply:
Yea, this `@include` actually applies to the entire `search` (so we don't make a meaningless call to Gravity for an empty term).